Node/core/lib/bots/ChatConnector.js

```diff
@@ -662,7 +662,10 @@ var ChatConnector = (function () {
                                 if (!refresh && _this.settings.appId && _this.settings.appPassword) {
                                     _this.authenticatedRequest(options, callback, true);
                                 }
-                                else {
```

Comment:
made this change conservatively - if after retrying to refresh the token we have an error prop on the response body, pass that error back to the bot error handler. else, do exactly what we were doing before.